path_login.go

```diff
@@ -213,12 +217,13 @@ func unauthorizedLogicalResponse(req *logical.Request, logger log.Logger, err er
 	return logical.RespondWithStatusCode(nil, req, http.StatusUnauthorized)
 }
 
-func requestTargetToMethodURL(requestTarget []string, expectedMethod string, expectedUrl string) (method string, url string, err error) {
+func requestTargetToMethodURL(requestTarget []string, expectedMethod string, expectedURLRegex string) (method string, url string, err error) {
 	if len(requestTarget) == 0 {
 		return "", "", errors.New("no (request-target) specified in header")
 	}
 	parts := strings.FieldsFunc(requestTarget[0], unicode.IsSpace)
-	if len(parts) != 2 || strings.ToLower(parts[0]) != expectedMethod || strings.ToLower(parts[1]) != expectedUrl {
+	match, err := regexp.MatchString(expectedURLRegex, strings.ToLower(parts[1]))
```

Comment:
Can `parts[1]` ever panic?

Reply:
Yes, it could've. Good catch. I've updated the approach in 4b0bd35.